includes/class-wc-stripe-helper.php

```
@@ -1105,6 +1105,57 @@ public static function has_cart_or_checkout_on_current_page() {
+			if ( class_exists( 'WC_Subscriptions_Product' ) && WC_Subscriptions_Product::is_subscription( $product ) ) {
+				return false;
+			}
+
+			// Pre-order (charge upon release) is not supported with adaptive pricing.
+			if ( class_exists( 'WC_Pre_Orders_Product' ) && WC_Pre_Orders_Product::product_is_charged_upon_release( $product ) ) {
+				return false;
+			}
+		}
+
+		return true;
+	}
```

PHPMD (2.15.0)

[warning] The method is_adaptive_pricing_supported() has a Cyclomatic Complexity of 14. The configured cyclomatic complexity threshold is 10. (undefined)

(CyclomaticComplexity)